library/kani/src/concrete_playback.rs

```diff
@@ -49,7 +51,11 @@ pub(crate) unsafe fn any_raw_internal<T, const SIZE_T: usize>() -> T {
     let mut next_concrete_val: Vec<u8> = Vec::new();
     CONCRETE_VALS.with(|glob_concrete_vals| {
         let mut_ref_glob_concrete_vals = &mut *glob_concrete_vals.borrow_mut();
-        next_concrete_val = mut_ref_glob_concrete_vals.pop().expect("Not enough det vals found");
+        next_concrete_val = if SIZE_T > 0 {
+            mut_ref_glob_concrete_vals.pop().expect("Not enough det vals found")
+        } else {
+            vec![]
```

Comment:
Does an empty array work? Not a big deal though.

Reply:
I don't think so. If I got your idea correctly, I don't think an array with defined size will type check with the variable-length case for non-zst. For slice, I think the slice's parent dies inside that closure so borrow checker complains. We need to return something that owns the memory.